kafka/protocol/metadata.py

```diff
@@ -47,6 +47,30 @@ class MetadataResponse_v1(Struct):
     )
 
 
+class MetadataResponse_v2(Struct):
```

Comment:
It's great that you added this, but maybe it should be in a dedicated commit? It just feels logically separate, and I'm unclear why it's needed for the `Fetch_Request/Response` stuff. Personally prefer a commit that says "Here's this new struct, here's what it changes compared to the old one" just because often spelunking in code I'll lookup the commit history to understand changes like this.